jib-plugins-common/src/main/java/com/google/cloud/tools/jib/plugins/common/ConfigurationPropertyValidator.java

```diff
@@ -128,5 +134,59 @@ public static ImageReference getGeneratedTargetDockerTag(
     }
   }
 
+  /**
+   * Parses a string in the form of "key1=value1,key2=value2,..." into a map.
+   *
+   * @param property the map string to parse, with entries separated by "," and key-value pairs
+   *     separated by "="
+   * @return the map of parsed values
+   */
+  public static Map<String, String> parseMapProperty(String property) {
+    // Matches key-value pairs in the form of "key=value"
+    Pattern environmentPattern = Pattern.compile("(?<name>[^=]+)=(?<value>.*)");
+    Map<String, String> result = new HashMap<>();
+
+    // Split on non-escaped commas
+    List<String> entries = parseListProperty(property);
+    for (String entry : entries) {
+      Matcher matcher = environmentPattern.matcher(entry);
+      if (!matcher.matches()) {
+        throw new IllegalArgumentException("'" + entry + "' is not a valid key-value pair");
+      }
+      result.put(matcher.group("name"), matcher.group("value"));
+    }
+    return result;
+  }
+
+  /**
+   * Parses a comma-separated string into a list. Ignores commas escaped with "\".
+   *
+   * @param property the comma-separated string
+   * @return the list of parsed values
+   */
+  public static List<String> parseListProperty(String property) {
+    List<String> result = new ArrayList<>();
+    boolean evenConsecutiveEscapes = true;
```

Comment:
I think this may be clearer if called `shouldEscape`? (and have it be inverted)

And then the loop would be like:
```
loop {
  if shouldEscape, then shouldEscape=false and skip
  if character is comma, do the logic
  if character is backslash, shouldEscape=true
}
```